File: KnapSack.py
def KnapSack(values, weights, sum):

    n = len(weights)

    dp=[[0 for x in range(sum+1)] for y in range(n+1)]

    for i in range(n+1):
        for j in range(sum+1):
            if i == 0 or j == 0:
                dp[i][j] = 0
            elif j >= weights[i-1]:
                dp[i][j] = max(dp[i-1][j], values[i-1] + dp[i-1][j-weights[i-1]])
            else:
                dp[i][j] = dp[i-1][j]
    return dp[n][sum]


values = [60, 100, 120]
weights = [10, 20, 30]
capacity = 50

result = KnapSack(values, weights, capacity)
print(result)

File: KnapSack.java
public class KnapSack {
public static int knapsack(int[] values, int[] weights, int sum){

        int n = weights.length;

        int[][] dp = new int[n+1][sum+1];

        for(int i = 0; i < n; i++) {
                dp[0][i] = 0;
        }


        for(int i = 1; i <=n; i++) {
                for(int j = 1; j <= sum; j++) {
                        if(weights[i-1] <= j) {
                                dp[i][j] = Math.max(dp[i-1][j], dp[i-1][j-weights[i-1]] + values[i-1]);
                        }

                }

        }
        return dp[n][sum];

}

public static void main(String[] args){

        int[] values = {60, 100, 120};
        int[] weights = {10, 20, 30};
        int capacity = 50;

        System.out.println(knapsack(values, weights, capacity));
}


}
